src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1302,6 +1331,70 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                 }
             }
 
+            private static string ReplaceUnknownEntriesWithSinkTag(DocumentsOperationContext context, ref string changeVector)
+            {
+                var globalDbIds = context.LastDatabaseChangeVector?.ToChangeVectorList()?.Select(x => x.DbId).ToList();
+                var incoming = changeVector.ToChangeVectorList();
+                var knownEntries = new List<ChangeVectorEntry>();
+                var newIncoming = new List<ChangeVectorEntry>();
+
+                foreach (var entry in incoming)
+                {
+                    if (globalDbIds?.Contains(entry.DbId) == true)
+                    {
+                        newIncoming.Add(entry);
+                        knownEntries.Add(entry);
+                    }
+                    else
+                    {
+                        newIncoming.Add(new ChangeVectorEntry
+                        {
+                            DbId = entry.DbId,
+                            Etag = entry.Etag,
+                            NodeTag = ChangeVectorExtensions.SinkTag
+                        });
+                    }
+                }
+
+                changeVector = newIncoming.SerializeVector();
+
+                return knownEntries.Count > 0 ? 
+                    knownEntries.SerializeVector() : 
+                    null;
+            }
+
+            private static void ReplaceKnownSinkEntries(DocumentsOperationContext context, ref string changeVector)
```

Comment:
Why do you need this?

Reply:
consider the following: (A - sink, B - hub)

1. doc create on sink (A:1- ... )
2. replicated to hub (SINK:1- ..)
3. modified on the the hub (SINK:1- .., B:1- ... )
4. replicated back to the sink (A:1-.., B:1-..)

I would like to avoid the tag SINK on the origin.

Reply:
We compare the db id, so no issue there.
And not a big worry from our perspective, because sink B will have the SINK from node A anyway, no?

Reply:
@karmeli87 ?

Reply:
I would like to avoid a situation where the sink node have the `SINK` tag on its own dbid.
I agree it is more a cosmetic issue, since we compare dbids, but it is "cleaner".